ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -252,27 +248,41 @@ export function AssetPickerModal({
       const blockedTokens = [];
 
       // Yield multichain tokens with balances
-      if (isMultiselectEnabled) {
-        for (const token of multichainTokensWithBalance) {
-          if (shouldAddToken(token.symbol, token.address, token.chainId)) {
-            yield token;
-          }
+      for (const token of multichainTokensWithBalance) {
+        if (shouldAddToken(token.symbol, token.address, token.chainId)) {
+          yield token.isNative
+            ? {
+                ...token,
+                image:
+                  CHAIN_ID_TOKEN_IMAGE_MAP[
+                    token.chainId as keyof typeof CHAIN_ID_TOKEN_IMAGE_MAP
+                  ],
+                type: AssetType.native,
+              }
+            : {
+                ...token,
+                // The Send flow requires the balance to be in Hex
+                balance: Numeric.from(token.balance ?? '0', 10)
+                  .shiftedBy(-1 * token.decimals)
+                  .toPrefixedHexString(),
+              };
```

Comment:
TIL the `yield` keyword 😅 

Probably a good thing to start adding elsewhere in our codebase to improve perf.